shared/js/background/chrome-events.es6.js

```diff
@@ -217,6 +219,17 @@ chrome.webNavigation.onCommitted.addListener(details => {
             const whitelisted = settings.getSetting('whitelisted')
             const tabURL = new URL(details.url) || {}
             if (!whitelisted || !whitelisted[tabURL.hostname]) {
+                // Set variables, which are used in the fingerprint-protection script.
+                const variableScript = {
+                    'code': `
```

Comment:
is this injected to the page context or extension's "isolated world" i.e. will this variable be visible to the page? Ideally it shouldn't.

Reply:
It will be visible to the page. Unfortunately, it's the only way I have found to pass variables in to the on page script before all other scripts load. I could potentially make the variable name more generic, and maybe obfuscate a few more things. Unfortunately, this runs into some of the same issues I hit with fingerprinting defense - any script we inject can potentially be observed by the page itself.

Reply:
Gotcha. Detectability is one thing, but I'm worried about name conflicts with variables defined by the page - let's make that variable name a bit longer/complicated.